src/main/java/com/intel/genomicsdb/GenomicsDBImporter.java

```diff
@@ -788,15 +701,17 @@ static File printLoaderJSONFile(
   /**
    * Creates a sorted list of callsets and generates unique TileDB
    * row indices for them. Sorted to maintain order between
-   * distributed share-nothing load processes.
+   * distributed share-nothing load processes. This method is synchronized
+   * to block multiple invocations (if by any chance) disturb the order in
+   * which TileDB row indexes are generated
    *
    * @param sampleNames list of sample names
    * @param useSamplesInOrderProvided  If True, do not sort the samples,
    *                                   use the order they appear in
    * @param lbRowIdx Smallest row idx which should be imported by this object
    * @return  Mappings of callset (sample) names to TileDB rows
    */
-  public static GenomicsDBCallsetsMapProto.CallsetMappingPB generateSortedCallSetMap(
+  public static synchronized GenomicsDBCallsetsMapProto.CallsetMappingPB generateSortedCallSetMap(
```

Comment:
But this method isn't updating any shared structures right? Even if 2 threads call this function in parallel, there should be no errors. Even with synchronized, there is no way of guaranteeing uniform order.